File: 프로그래머스/lv0/120923. 연속된 수의 합/연속된 수의 합.java
import java.util.Arrays;

class Solution {
    public int[] solution(int num, int total) {
        int[] answer = {};
        
        // a, a+d, a+2d, a+3d, a+4d, ... , a+(num-1)d = total
        // num * a + num*(num+1)/2 = total
        // num * a = total - num*(num+1)/2
        // a = (total-num*(num+1)/2)/num
        
        int a = (total-num*(num+1)/2)/num;
        int[] arr = new int[num];
        
        
        for(int i=0; i<num; i++) arr[i] = a + i + 1;
        answer = arr;
        
        
        
        return answer;
    }
}

File: 프로그래머스/lv0/120923. 연속된 수의 합/연속된 수의 합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int num, int total) {
    vector<int> answer;
    
    // a, a+d, a+2d, a+3d, a+(n-1)d
    // = a*n + d+2d+3d+..(n-1)d
    // = a*n + (n*(n-1)/2)*d
    // = total
    // d = 1
    // a = (total - n(n-1)/2) / n
    int a = (total - num*(num-1)/2) / num;
    for(int i=0; i<num; i++){
        answer.push_back(a+i);
    }
    
    return answer;
}